cupy/core/fusion.py

```diff
@@ -340,240 +538,186 @@ def res(*args, **kwargs):
+            in_types (list of dtypes): The list of dtypes of input parameters.
+            name (str): The name of the kernel.
+
+        Return value (tuple of ElementwiseKernel/ReductionKernel and dict):
+            The second element of return values is kwargs that will give into
+            the elementwise kernel or reduction kernel.
+        """
+        in_params = [self.fresh_premap_param(t) for t in in_dtypes]
+        in_pvars = [FusionVarPython(_, False) for _ in in_params]
+        out_pvars = func(*in_pvars)
+        if isinstance(out_pvars, tuple):
+            out_pvars = list(out_pvars)
+        else:
+            out_pvars = [out_pvars]
+        out_pvars = [_ for _ in out_pvars if _ is not None]
+        out_cvars = [self.get_cuda_var(_) for _ in out_pvars]
+
+        out_dtypes = [_.dtype for _ in out_pvars]
+        out_params = [self.fresh_premap_param(t) for t in out_dtypes]
+
+        in_params_code = ', '.join(var.declaration_param()
+                                   for var in in_params)
+        out_params_code = ', '.join(var.declaration_param()
+                                    for var in out_params)
+
+        operation = self.emit_operation_code()
+        submodule_code = self.get_submodules_code()
+
+        if self.reduce_op is None:
+            operation += ' '.join('{} = {};'.format(t, s)
+                                  for s, t in zip(out_cvars, out_params))
+            kernel = core.ElementwiseKernel(
+                in_params_code, out_params_code, operation,
+                preamble=submodule_code,
+                name=name)
+            return kernel, {}
+        else:
+            op = self.reduce_op
+            _, (fixed_type,), (_, reduce_code, fix_code, reduce_ctype) = op
+            if reduce_ctype is None:
+                reduce_ctype = 'type_in0_raw'
+
+            fixed_dtype = numpy.dtype(fixed_type)
+            fixed_ctype = _dtype_to_ctype[fixed_dtype]
+
+            postmap_code = '// {} operations\n'.format(
+                len(self.postmap_op_list))
+            postmap_code += ''.join(v.declaration()
+                                    for v in self.postmap_local_list)
+            postmap_code += ''.join(op.declaration_args()
+                                    for op in self.postmap_op_list)
+            postmap_code += ''.join(op.code() for op in self.postmap_op_list)
+            postmap_code += ' '.join('{} = {};'.format(t, s)
+                                     for s, t in zip(out_cvars, out_params))
+
+            submodule_code += self.emit_premap_code(in_params, operation)
+            submodule_code += 'typedef {} type_in0_raw;\n'.format(fixed_ctype)
+            submodule_code += 'typedef {} type_out0_raw;\n'.format(fixed_ctype)
+            submodule_code += self.emit_fix_code(
+                reduce_ctype, fixed_dtype, fix_code)
+            submodule_code += self.emit_postmap_code(out_params, postmap_code)
```

Comment:
What is "fix code"?
Is "fix" related to "fixed" in `fixed_dtype` etc.?
Please write somewhere.